src/main/scala/com/raphtory/test/allcommands/ResultsCompare.scala

```diff
@@ -0,0 +1,121 @@
+package com.raphtory.test.allcommands
+
+import java.net.InetAddress
+
+import com.mongodb.casbah.MongoClientURI
+import org.mongodb.scala.bson.codecs.Macros._
+import org.mongodb.scala.bson.codecs.DEFAULT_CODEC_REGISTRY
+import org.bson.codecs.configuration.CodecRegistries.{fromProviders, fromRegistries}
+import org.mongodb.scala.{MongoClient, MongoCollection, MongoDatabase}
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+
+
+object ResultsCompare extends App{
+  if(args.length<4) {
+    println("Please provide analyser job ID's to compare")
+    System.exit(1)
+  }
+  stateCheck(args(0),args(1))
+  connectedComponentsCheck(args(2),args(3))
+
+
+
+  //val toCompare = Await.result(database.getCollection(args(1)).find().toFuture(), Duration.Inf)
+  def stateCheck(baseCollectionID:String,compareCollectionID:String) = {
+    val dbname    = System.getenv().getOrDefault("ANALYSIS_MONGO_DB_NAME", "raphtory").trim
+    val codecRegistry = fromRegistries(fromProviders(classOf[StateCheckResult]), DEFAULT_CODEC_REGISTRY )
+    val mongo = MongoClient()
+    val database: MongoDatabase = mongo.getDatabase("raphtory").withCodecRegistry(codecRegistry)
+
+    val baseCollection: MongoCollection[StateCheckResult] = database.getCollection(baseCollectionID)
+    val base = Await.result(baseCollection.find().toFuture(), Duration.Inf).map(state=>(TimeParams(state.time,state.window),state)).toMap
+
+    val compareCollection: MongoCollection[StateCheckResult] = database.getCollection(compareCollectionID)
+    val compare = Await.result(compareCollection.find().toFuture(), Duration.Inf).map(state=>(TimeParams(state.time,state.window),state)).toMap
+
+    val stateResult = base.map(baseValue=> baseValue._2.compareTo(compare(baseValue._1)))
+    if(stateResult.fold(true){(x,y) =>x&&y})
```

Comment:
Can this be replaced with forall(_ == true)?

```
val list = List(true, true, false, true)
val res = list.forall(_ == true)
```

Reply:
I would like to integrate this class into some more automated testing, so yeah will include then.